### PR TITLE
feat: add initial host implementation

### PR DESCRIPTION
**Pull request by rvolosatovs**
*Please review commit-by-commit (https://github.com/wasmCloud/wasmCloud/pull/273/commits) and mind the commit messages/descriptions*

Crate documentation available at https://rvolosatovs.github.io/wasmCloud/wasmcloud/index.html

This PR contains:
- initial host implementation, which:
	- executes actors via wasmtime
	- implements the `wasmbus` FFI interface
	- contains built-in provider implementation adaptors for `log` and `rand` crates
- CI setup for linting, testing and build of cross-compiled binaries along with OCI images containing them

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [x] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [x] aarch64-linux
- [x] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)

Yes, see `tests` within `src/actor.rs`

### Acceptance or Integration

See `tests`

### Manual Verification

Tested with a custom `http-log-rng` actor utilizing builtin providers and https://github.com/wasmCloud/examples/tree/main/actor/echo integrated into this repository

### CODE REVIEW

**Review by autodidaddict — DISMISSED** [review]
@rvolosatovs and I have had some discussion about this codebase and we've got some ideas for how to refactor it and smooth out the devex going forward, but I don't see anything here that should block us from just iterating on main. 

LGTM to go ahead unless @ricochet has additional thoughts

**Comment by autodidaddict** [comment]
we're also going to want to add CI support, branch protection, etc.

**Comment by rvolosatovs** [comment]
> we're also going to want to add CI support, branch protection, etc.

The CI is already set up as part of the PR, the checks should just be enabled in the settings

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[toolchain]
+channel = "nightly"
+components = ["clippy", "rustfmt"]
+targets = [
+    "wasm32-unknown-unknown"
```
Comment (ricochet) [inline rust-toolchain.toml:5]: wasm32-wasi?

Comment (rvolosatovs) [reply rust-toolchain.toml:5]: There's nothing we currently build for WASI here

**Review by ricochet — DISMISSED** [review]

**Comment by ricochet** [comment]
> LGTM to go ahead unless @ricochet has additional thoughts

My thoughts are "LETS GOOOOOOO"

**Comment by rvolosatovs** [comment]
@autodidaddict should I change the target branch to `main` or keep it at `dev`?

**Review by autodidaddict — APPROVED** [review]
Let's just make sure that as we go forward we don't lose the badges and other summary information on the root README.

**Review by ricochet — APPROVED** [review]

**Pull request merged**